pyteal/ast/router_test.py

```diff
@@ -422,35 +429,55 @@ def test_wrap_handler_method_call():
             )
         ]
 
-        loading: list[pt.Expr]
+        app_args = [
+            arg for arg in args if arg.type_spec() not in pt.abi.TransactionTypeSpecs
+        ]
+
+        app_arg_cnt = len(app_args)
+
+        txn_args = [
+            arg for arg in args if arg.type_spec() in pt.abi.TransactionTypeSpecs
+        ]
 
-        if abi_subroutine.subroutine.argument_count() > pt.METHOD_ARG_NUM_CUTOFF:
+        loading: list[pt.Expr] = []
+
+        if app_arg_cnt > pt.METHOD_ARG_NUM_CUTOFF:
             sdk_last_arg = pt.abi.TupleTypeSpec(
-                *[
-                    spec
-                    for spec in typing.cast(
-                        list[pt.abi.TypeSpec],
-                        abi_subroutine.subroutine.expected_arg_types,
-                    )[pt.METHOD_ARG_NUM_CUTOFF - 1 :]
-                ]
+                *[arg.type_spec() for arg in app_args[pt.METHOD_ARG_NUM_CUTOFF - 1 :]]
             ).new_instance()
+
             loading = [
                 arg.decode(pt.Txn.application_args[index + 1])
-                for index, arg in enumerate(args[: pt.METHOD_ARG_NUM_CUTOFF - 1])
+                for index, arg in enumerate(app_args[: pt.METHOD_ARG_NUM_CUTOFF - 1])
             ]
+
             loading.append(
                 sdk_last_arg.decode(pt.Txn.application_args[pt.METHOD_ARG_NUM_CUTOFF])
             )
-            for i in range(pt.METHOD_ARG_NUM_CUTOFF - 1, len(args)):
-                loading.append(
-                    sdk_last_arg[i - pt.METHOD_ARG_NUM_CUTOFF + 1].store_into(args[i])
-                )
         else:
             loading = [
                 arg.decode(pt.Txn.application_args[index + 1])
-                for index, arg in enumerate(args)
+                for index, arg in enumerate(app_args)
             ]
 
+        if len(txn_args) > 0:
```

Comment:
This test code has code that mirrors the implementation very closely, which isn't great. I would have more confidence if you added a test with very little logic, similar to `test_wrap_handler_method_call_many_args` below -- that was added to give us more confidence that were were testing tupling arguments appropriately.

Reply:
Yes, the new feature related to transaction types